cache/src/main/scala/coursier/Cache.scala

```diff
@@ -975,6 +984,29 @@ object Cache {
       case (f, None) => EitherT(S.point[Either[FileError, File]](Right(f)))
       case (f, Some(c)) =>
         validateChecksum(artifact, c, cache, pool).map(_ => f)
+    }.leftFlatMap {
+      case err: FileError.WrongChecksum =>
+        if (retry <= 0) {
+          EitherT(S.point(Left(err)))
+        }
+        else {
+          val badFile = localFile(artifact.url, cache, artifact.authentication.map(_.user))
+          badFile.delete()
```

Comment:
This call should be wrapped in a `S.schedule(pool) { … }`, as it's doing blocking stuff, like
```scala
S.schedule(pool) {
  badFile.delete()
  logger.foreach(…)
}.flatMap { _ =>
  file(…)
}
```

Reply:
done